### PR TITLE
Refactor multi line into multi line logic

### PR DESCRIPTION
**Pull request by bazsi**
This series of patches refactors the multi-line related logic into a separate class, instead of being embedded into LogProtoMultiLineServer and related classes.

As such it is in preparation for creating a parser that combines multi-line messages into a single line (#4225)

This should not have a functionality impact, apart from the removal of an undocumented transport() option, as it was possible to request the indentation based multi-line format using transport(intented-multi-line), e.g. this worked in the past:

source { network(transport(indented-multi-line)); };

Which will not work from this point on. This was never documented. If we want to support multi-line formats for network sources, we would need to expose multi-line related options in the grammar of network/tcp/etc. And that would make it possible to combine various transports and multi-line processing, which were not be possible with the trick above. We could restore this functionality but I don't think it is really used anywhere.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/affile/logproto-file-reader.c [hunk 1]
```diff
@@ -30,56 +30,72 @@ LogProtoServer *
...
+log_proto_file_reader_options_init(LogProtoFileReaderOptions *options, GlobalConfig *cfg)
 {
-  log_proto_multi_line_server_options_init(&options->super);
-  return log_proto_file_reader_options_validate(options);
-}
+  if (!log_proto_file_reader_options_validate(options))
+    return FALSE;
 
-void
-log_proto_file_reader_options_destroy(LogProtoFileReaderOptions *options)
```
Comment (alltilla) [inline modules/affile/logproto-file-reader.c:82]: [optional]

This can be removed from the header, too.

**Comment by alltilla** [comment]
Rebased to master.

**Review by alltilla — APPROVED** [review]
LGTM, approve!

It would be nice to see the ALTP tests being green with these commits. @HofiOne Could you kindly run the ALTP tests on this branch? Thanks!

**Comment by HofiOne** [comment]
> LGTM, approve!
> 
> It would be nice to see the ALTP tests being green with these commits. @HofiOne Could you kindly run the ALTP tests on this branch? Thanks!

@bazsi @alltilla I guess one of these changes broke the macOS builds
https://github.com/syslog-ng/syslog-ng/actions/runs/4071634286/jobs/7013609453#step:7:223
it seems MongoDB option is now forced to be ON always however it must be OFF for macOS currently AFAIK.
(or, if these changes contain a fix for MongoDB for macOS as well, we just miss the corresponding library on the macOS test machine)

**Comment by alltilla** [comment]
@HofiOne Is it possible to run the ALTP tests on this branch now? It would be nice to see them pass, but if you don't mind merging this without testing, then it is okay for me, too.

**Comment by HofiOne** [comment]
> @HofiOne Is it possible to run the ALTP tests on this branch now? It would be nice to see them pass, but if you don't mind merging this without testing, then it is okay for me, too.

sorry, what do you exactly mean by ALTP tests?

**Comment by alltilla** [comment]
ALTP is a PE only reliable transport. It has extensive testdb tests.
If you run full PE installer tests, ALTP test cases will be run, too.

**Comment by alltilla** [comment]
@kira-syslogng retest this please

**Comment by HofiOne** [comment]
> ALTP is a PE only reliable transport. It has extensive testdb tests. If you run full PE installer tests, ALTP test cases will be run, too.

sorry, I was not specific enough, can you direct me to the specific test cases to run as running full tests for all platforms could take a long time, or the random platform builds are well enough now?

**Comment by alltilla** [comment]
Sorry, I do not remember the testdb directory structure :( If I remember correctly there are `altp` tag on those tests, but I am not sure.

Random platform tests are completely enough.

I am sorry about the network-based-driver tests fails :(

**Comment by HofiOne** [comment]
> Sorry, I do not remember the testdb directory structure :( If I remember correctly there are `altp` tag on those tests, but I am not sure.
> 
> Random platform tests are completely enough.
> 
> I am sorry about the network-based-driver tests fails :(

I'll run a random platform test tonight for this branch, thanks for the confirmation!

**Comment by HofiOne** [comment]
the tests are finished, we still have a lot of failing test cases but I cannot see anything related to ALTP, so I think we are fine
(just to be sure, if you wish I can share with you the complete list of the failing tests privately to double-check if I was wrong or right)

**Review by alltilla — APPROVED** [review]

**Comment by alltilla** [comment]
> the tests are finished, we still have a lot of failing test cases but I cannot see anything related to ALTP, so I think we are fine (just to be sure, if you wish I can share with you the complete list of the failing tests privately to double-check if I was wrong or right)

Thank you for running the tests. I believe, you have checked thoroughly, which test cases failed, there is no need to double check it. :) 

I think it is safe to merge this PR. If you think the same, please do so, it is approved. Thanks!

**Pull request merged**